src/Workspaces/Core/Portable/Workspace/Workspace_Editor.cs

```diff
@@ -435,7 +435,7 @@ private ISet<DocumentId> GetProjectOpenDocuments_NoLock(ProjectId project)
                     // Note: we pass along the newText here so that clients can easily get the text
                     // of an opened document just by calling TryGetText without any blocking.
                     currentSolution = oldSolution.WithDocumentTextLoader(documentId,
-                        new ReuseVersionLoader(oldDocument.State, newText), newText, PreservationMode.PreserveIdentity);
+                        new ReuseVersionLoader((DocumentState)oldDocument.State, newText), newText, PreservationMode.PreserveIdentity);
```

Comment:
There has to e a way to do this in a typesafe manner.   Why can't oldDocument.State still return a DocumentState?